### PR TITLE
Create keyspaces for all storage groups if * domain is provided

### PR DESCRIPTION
**Pull request by Pchelolo**
Now in case the request to create bucket came with a `*` domain, it will create the bucket for all the defined storage groups.

cc @wikimedia/services

### CODE REVIEW

File: lib/db.js [hunk 1]
```diff
@@ -177,14 +181,24 @@ class DB {
      * @return {string}         Valid Cassandra keyspace key
      */
     keyspaceName(domain, table) {
-        const cacheKey = JSON.stringify([domain, table]);
+        const cacheKey = JSON.stringify(['domain', domain, table]);
```
Comment (d00rman) [inline lib/db.js]: I think this should be optimised. We don't need a per-domain cache, only on a per-storage-group basis.

**Comment by d00rman** [comment]
Oh, and the version should be bumped

File: lib/db.js (hunk 1 above)
Comment (Pchelolo) [reply lib/db.js]: mmm... maybe this is an overoptimization, but we now have both by-domain and by-group caching. Resolving a storage group could mean executing quite a lot of regexes..

Comment (d00rman) [reply lib/db.js]: Oh, right. What I meant was that the domains cache should be the mapping to the storage group and from then onward all the cache lookups should be by storage group. Instead, I believe the caches now map domains directly to names and statements, which means more executions than really needed.

Comment (Pchelolo) [reply lib/db.js]: mmm.. we didn't really change how the caches work, we only introduced one more cache `storageGroup => ksName` that we didn't have before. If I understand correctly what you're proposing is to change all other caches to be `storageGroup => whatever` - it's a good optimization, but not really a part of this PR

**Comment by d00rman** [comment]
Published as [v1.2.0](https://github.com/wikimedia/restbase-mod-table-cassandra/tree/v1.2.0)

**Pull request merged**